lib/tools/apk-utils.js

```diff
@@ -53,19 +53,30 @@ apkUtilsMethods.isAppInstalled = async function isAppInstalled (pkg) {
  *
  * @param {string} uri - The name of URI to start.
  * @param {string} pkg - The name of the package to start the URI with.
+ * @param {boolean} waitForLaunch - Optional parameter of waitForLaunch caps to add wait in adb command
  */
-apkUtilsMethods.startUri = async function startUri (uri, pkg) {
+apkUtilsMethods.startUri = async function startUri (uri, pkg, waitForLaunch = true) {
+  let args;
   if (!uri || !pkg) {
     throw new Error('URI and package arguments are required');
   }
 
-  const args = [
-    'am', 'start',
-    '-W',
-    '-a', 'android.intent.action.VIEW',
-    '-d', uri.replace(/&/g, '\\&'),
-    pkg,
-  ];
+  if (!waitForLaunch) {
+    args = [
```

Comment:
Please do not duplicate args. The `-W` could be just pushed based on the provided options:

```
const args = ['am', 'start'];
if (waitForLaunch) {
  args.push('-W');
}
args.push('-a', 'android.intent.action.VIEW',
       '-d', uri.replace(/&/g, '\\&'),
       pkg);
```

Reply:
@mykola-mokhnach Thanks. I will do the suggested changes.

Reply:
const args = ['am', 'start'];
  if (opts.waitForLaunch) {
    args.push('-W');
  }
  args.push('-a', 'android.intent.action.VIEW',
       '-d', uri.replace(/&/g, '\\&'),
       pkg);